Quaver.Shared/Screens/ConfirmDialog.cs

```diff
@@ -0,0 +1,236 @@
+        /// </summary>
+        public Sprite ContainingBox { get; set; }
+
+        /// <summary>
+        ///     The top line of the containing box.
+        /// </summary>
+        public Line TopLine { get; set; }
+
+        /// <summary>
+        ///     Text that asks the user if they're sure they want to quit.
+        /// </summary>
+        public SpriteText AreYouSure { get; set; }
+
+        /// <summary>
+        ///     The line that divides AreYouSure and the buttons.
+        /// </summary>
+        public Line DividerLine { get; set; }
+
+        /// <summary>
+        ///     The button to confirm the action.
+        /// </summary>
+        public TextButton ConfirmButton { get; set; }
+
+        /// <summary>
+        ///     The button to cancel the dialog.
+        /// </summary>
+        public TextButton CancelButton { get; set; }
+
+        /// <summary>
+        ///     Event to verify confirmation.
+        /// </summary>
+        public event EventHandler<bool> Confirmation;
+
+        private string AreYouSureText { get; }
+
+        /// <inheritdoc />
+        /// <summary>
+        /// </summary>
+        public ConfirmDialog(string areYouSureText) : base(0.75f)
+        {
+            Logger.Debug($"Opened ConfirmDialog", LogType.Runtime);
+
+            AreYouSureText = areYouSureText;
+
+            CreateContent();
+        }
+
+        /// <inheritdoc />
+        /// <summary>
+        /// </summary>
+        public sealed override void CreateContent()
+        {
+            CreateContainingBox();
+            CreateTopLine();
+            CreateAreYouSureText();
+            CreateDividerLine();
+            CreateConfirmButton();
+            CreateCancelButton();
+        }
+
+        /// <summary>
+        /// </summary>
+        /// <param name="gameTime"></param>
+        public override void Update(GameTime gameTime)
+        {
+            var dt = gameTime.ElapsedGameTime.TotalMilliseconds;
+
+            ConfirmButton.Border.FadeToColor(ConfirmButton.IsHovered ? Color.White : Color.Crimson, dt, 60);
+            ConfirmButton.Text.FadeToColor(ConfirmButton.IsHovered ? Color.White : Color.Crimson, dt, 60);
+            CancelButton.Border.FadeToColor(CancelButton.IsHovered ? Color.White : Colors.MainAccent, dt, 60);
+            CancelButton.Text.FadeToColor(CancelButton.IsHovered ? Color.White : Colors.MainAccent, dt, 60);
+
+            base.Update(gameTime);
+        }
+
+        /// <inheritdoc />
+        /// <summary>
+        /// </summary>
+        /// <param name="gameTime"></param>
+        public override void HandleInput(GameTime gameTime)
+        {
+            if (KeyboardManager.IsUniqueKeyPress(Keys.Escape))
+                Dismiss("KeyboardManager (ESC Button)");
+
+            if(KeyboardManager.IsUniqueKeyPress(Keys.Enter))
+                Confirm("Keyboard Manager (Enter Button)");
+        }
+
+        private void Confirm(string via = "")
```

Comment:
I don't think the via is necessary anymore. It was only used during debugging, and I forgot to remove it.